server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSCreateSPOptionsCheckUtility.java

```diff
@@ -117,5 +128,32 @@ private void checkOpts(CICSParser.Cics_create_optsContext ctx) {
         }
         checkHasMutuallyExclusiveOptions("LOG or NOLOG or LOGMESSAGE", ctx.LOG(), ctx.NOLOG(), ctx.LOGMESSAGE());
     }
+
+ private void checkDataValueCompleteDiscard(CICSParser.Cics_create_optsContext ctx) {
+        if (ctx.children == null) return;
+        for (int index = 0; index < ctx.children.size() - 1; index++) {
+            if (!TerminalNode.class.isAssignableFrom(ctx.children.get(index).getClass())
+                || !CICSParser.Cics_data_valueContext.class.isAssignableFrom(ctx.children.get(index + 1).getClass())) continue;
+           int tokenIndex = ((TerminalNode) ctx.children.get(index)).getSymbol().getType();
+            if (Arrays.stream(COMMANDS_WITH_DISCARD_COMPLETE_OPTS).anyMatch(i -> i == tokenIndex)
+        ) {
+                throwBrowsingViolation(
+                        ctx.children.get(index),
+                        "Sub Operand");
+            }
+        }
+    }
+    private void checkRequiredSubOperand(CICSParser.Cics_create_optsContext ctx) {
+        if (ctx.children == null) return;
+        for (int index = 0; index < ctx.children.size() - 1; index++) {
+            if (!TerminalNode.class.isAssignableFrom(ctx.children.get(index).getClass())
+                    || CICSParser.Cics_data_valueContext.class.isAssignableFrom(ctx.children.get(index + 1).getClass())) continue;
```

Comment:
As discussed, this code fails to generate a diagnostic when the operand in the last token in the context.